### PR TITLE
LegacyScans: Migrate to HttpSource

### LINKED ISSUE #1741: Legacy Scans : no series list
### Source information

Legacy Scans 1.4.30

### Source language

French

### Steps to reproduce

1. go to Explorer > Source 
2. open Legacy Scans 

### Expected behavior

This should display the series list

### Actual behavior

But it shows us that "No results found".
unnamed_3
And yet, in WebView, the site works without error.
unnamed_4

### Mihon/Tachiyomi version

0.15.3

### Android version

Android 13

### Other details

Likewise in Lastet and Filter
unnamed_6
unnamed_5

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by choppeh**
Closes #1741

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt [hunk 1]
```diff
@@ -1,9 +1,202 @@
...
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    private val json: Json by injectLazy()
+
+    override fun popularMangaRequest(page: Int) = GET("$apiUrl/misc/views/all", headers)
+
+    override fun popularMangaParse(response: Response) =
+        mangasPageParse(response.parseAs<List<MangaDto>>()).copy(hasNextPage = false)
```
Comment (AwkwardPeak7) [inline src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt]: pass an optional parameter to `mangasPageParse` to set `hasNextPage` so copying can be avoided

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt [hunk 2]
```diff
@@ -1,9 +1,202 @@
...
+    override fun latestUpdatesRequest(page: Int): Request {
+        val offset = pageOffset(page)
+        val url = "$apiUrl/misc/comic/home/updates".toHttpUrl().newBuilder()
+            .addQueryParameter("start", "${offset.first}")
+            .addQueryParameter("end", "${offset.second}")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage = popularMangaParse(response)
```
Comment (AwkwardPeak7) [inline src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt]: you passed offsets to the request, does it still not have next page?

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt [hunk 3]
```diff
@@ -1,9 +1,202 @@
...
+            val manga = SManga.create().apply {
+                url = "/comics/${query.substringAfter(URL_SEARCH_PREFIX)}"
+            }
+            client.newCall(mangaDetailsRequest(manga)).asObservableSuccess().map { response ->
+                val document = Jsoup.parse(response.peekBody(Long.MAX_VALUE).string())
+                // Avoid null pointer
+                when {
+                    isMangaPage(document) -> {
+                        MangasPage(listOf(mangaDetailsParse(response)), false)
+                    }
```
Comment (AwkwardPeak7) [inline src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt]: create a `mangaDetailsParse` which takes `Document` so response copying can be avoided

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt [hunk 4]
```diff
@@ -1,9 +1,202 @@
...
+        return start to end
+    }
+
+    private fun mangasPageParse(dto: List<MangaDto>): MangasPage {
+        val mangas = dto.map {
+            SManga.create().apply {
+                title = it.title
+                if (it.isCoverPresent()) {
+                    thumbnail_url = "$apiUrl/${it.cover}"
+                }
```
Comment (AwkwardPeak7) [inline src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt]:
```suggestion
                thumbnail_url = it.cover?.let { cover -> "$apiUrl/$cover" }
```

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScansFilter.kt [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    "C",
+    "m",
+    "b",
+    "S",
+    "h",
+    "Smut",
+    "démons",
+    "655bbef5f0599f6198dd7b73",
+    "Virtuel world",
+    "665d02bd6356105ec136902f",
```
Comment (AwkwardPeak7) [inline src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScansFilter.kt]: these random strings are genres..?

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt (hunk 2 above)
Comment (choppeh) [reply src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt]: Fixed

File: src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScansFilter.kt (hunk 5 above)
Comment (choppeh) [reply src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScansFilter.kt]: Oh, I forgot to clean that too.

**Review by AwkwardPeak7 — APPROVED** [review]

**Comment by Mourad-Saidomar** [comment]
Thanks😄

**Pull request merged**